ccl/streamingccl/streamingest: TestStreamingRegionalConstraint failed

ccl/streamingccl/streamingest.TestStreamingRegionalConstraint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11802562?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11802562?buildTab=artifacts#/) on master @ [985662236d7bf273b93a7b5e32def8e2d1043640](https://github.com/cockroachdb/cockroach/commits/985662236d7bf273b93a7b5e32def8e2d1043640):


```
=== RUN   TestStreamingRegionalConstraint
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/90c1b75d835f45b8488807abb5b1092d/logTestStreamingRegionalConstraint3377522745
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/streamingccl/replicationtestutils/testutils.go:314: (startC2CTestCluster)
	NOTICE: .Clock() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Clock() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization

pkg/ccl/streamingccl/replicationtestutils/testutils.go:122: ((*TenantStreamingClusters).setupSrcTenant)
	NOTICE: .StartSharedProcessTenant() called via implicit interface TenantControlInterface;
HINT: consider using .TenantController().StartSharedProcessTenant() instead.
    replication_stream_e2e_test.go:1216: condition failed to evaluate within 45s: from replication_stream_e2e_test.go:1202: running iterate callback: found table data located on another node 3
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/90c1b75d835f45b8488807abb5b1092d/logTestStreamingRegionalConstraint3377522745
--- FAIL: TestStreamingRegionalConstraint (149.40s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStreamingRegionalConstraint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31596

--- a/pkg/cmd/teamcity-trigger/main.go
+++ b/pkg/cmd/teamcity-trigger/main.go
@@ -159,7 +159,8 @@ func runTC(queueBuild func(string, map[string]string)) {
 
 		if testTarget == "//pkg/sql/logictest:logictest_test" ||
 			testTarget == "//pkg/kv/kvserver:kvserver_test" ||
-			testTarget == "//pkg/ccl/backupccl:backupccl_test" {
+			testTarget == "//pkg/ccl/backupccl:backupccl_test" ||
+			testTarget == "//pkg/ccl/streamingccl/streamingest:streamingest_test" {
 			// Stress heavy with reduced parallelism (to avoid overloading the
 			// machine, see https://github.com/cockroachdb/cockroach/pull/10966).
 			parallelism /= 2

